src/averageAges.js

```diff
@@ -55,7 +69,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    child.mother && people.some(parent => parent.name === child.mother));
```

Comment:
You can simplify the children filtering by checking if onlyWithSon is true and child.sex === 'm' in the same filter function.